vdm-stub-generator/src/main/groovy/com/anaplan/engineering/vdmstubgenerator/VdmStubGenerationTask.groovy

```diff
@@ -5,40 +5,57 @@ import com.anaplan.engineering.vdmanimation.api.VdmFile
 import com.anaplan.engineering.vdmanimation.api.VdmSpecification
 import org.gradle.api.DefaultTask
 import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.InputDirectory
-import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.*
 
 // TODO -- remove horrible hack to handle duplicate undef!
 abstract class VdmStubGenerationTask extends DefaultTask {
 
     @Input
     abstract Property<String> getTargetPackage()
 
-    @InputFile
-    abstract Property<File> getSpecificationManifest()
-
+    // Typically used for specification files located in a jar
+    @Optional
     @InputDirectory
-    abstract Property<File> getSpecificationDir()
+    abstract Property<File> getSpecificationDirectory()
+
+    // Typically used for local specification files
+    @Optional
+    @Input
+    abstract Property<String> getSpecificationManifest()
+
+    @OutputDirectory
+    abstract Property<File> getStubDirectory()
 
     @TaskAction
     def generateDefinitions() {
         def targetPackage = getTargetPackage().getOrElse("com.anaplan.engineering.vdmstubgenerator.generated")
-        def manifestFile = getSpecificationManifest().getOrElse(new File(project.projectDir,"build/generated-resources/vdm/specification.mf"))
-        def specDir = getSpecificationDir().getOrElse(new File(project.projectDir,"build/generated-resources/"))
-
-        int i = 0;
-        def specification = new Specification(files: manifestFile.readLines().collect { file ->
-            new VdmFile("${i++}.vdmsl", new File(specDir,"$file").text)
-        })
+        def vdmFiles = []
+        def manifestUrl = getClass().getResource(getSpecificationManifest().getOrElse("/vdm/specification.mf"))
+        if (manifestUrl != null) {
+            vdmFiles.addAll(manifestUrl.readLines().indexed().collect { i, file ->
+                    new VdmFile("${i++}.vdmsl", getClass().getResource("/$file").text)
```

Comment:
⛏️  Are you sure you want to increment the index here? This looks like a leftover from the old code. It looks out of place and a little surprising, but it just results in all the file numbers being shifted by 1 since `i` is a local variable.
```suggestion
                    new VdmFile("$i.vdmsl", getClass().getResource("/$file").text)
```